### PR TITLE
feat(clients/java): add the default region using the camunda cloud builder

### LINKED ISSUE #7391: Adjust Camunda Cloud client builder to new domain names
**Is your feature request related to a problem? Please describe.**

Starting with 1.1 Zeebe clusters in Camunda Cloud will be accessible under a new domain name. The domain name will now include the region of the cluster. For example the domain name `2bd0916e-6f5d-40fa-9d26-6d7d209a06ce.zeebe.camunda.io` will change to `2bd0916e-6f5d-40fa-9d26-6d7d209a06ce.bru-2.zeebe.camunda.io`.

**Describe the solution you'd like**

Per default the Camunda Cloud client builder should add the current default region `bru-2` to the domain. And allow users to specify the domain during building of the client.

### PR DESCRIPTION
**Pull request by menski**
## Description

- Adds `bru-2` as the default region to URLs build using the Camunda Cloud builder
- Adds a new method `withRegion` to the Camunda Cloud builder to allow users to override the region, i.e. if we have multiple region support
- Adds the region to the audience as it is used in the credentials cache

## Related issues

closes #7391 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientCloudBuilderStep1.java
Comment (npepinpe) [inline clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientCloudBuilderStep1.java:50]: We need to document what regions a user should put here, in the docs, or in console. Additionally, if we document it in the official docs page, I would also point it out here.

**Review by npepinpe — APPROVED** [review]
Thanks!

Please add the region to the builder impl's toString. I don't see a need for a second review pass here though.

File: clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientCloudBuilderStep1.java
Comment (menski) [reply clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientCloudBuilderStep1.java:50]: I created an docs issue for this https://github.com/camunda-cloud/camunda-cloud-documentation/issues/302. I guess currently we don't expose the region in way in Console.

**Pull request merged**
